GOVERNANCE.md

```diff
@@ -1,30 +1,46 @@
+* Express interest to the senior maintainers that your organization is interested in becoming a
+  maintainer. Becoming a maintainer generally means that you are going to be spending substantial
   time (>25%) on Envoy for the foreseeable future. You should have domain expertise and be extremely
-  proficient in C++. Ultimately your goal is to become a senior committer that will represent your
+  proficient in C++. Ultimately your goal is to become a senior maintainer that will represent your
   organization.
 * We will expect you to start contributing increasingly complicated PRs, under the guidance
-  of the existing senior committers.
+  of the existing senior maintainers.
 * We may ask you to do some PRs from our backlog.
 * As you gain experience with the code base and our standards, we will ask you to do code reviews
-  for incoming PRs (i.e., all committers are expected to shoulder a proportional share of
+  for incoming PRs (i.e., all maintainers are expected to shoulder a proportional share of
   community reviews).
 * After a period of approximately 2-3 months of working together and making sure we see eye to eye,
-  the existing senior committers will confer and decide whether to grant commit status or not. We
-  make no guarantees on the length of time this will take, but 2-3 months is the approximate
+  the existing senior maintainers will confer and decide whether to grant maintainer status or not.
+  We make no guarantees on the length of time this will take, but 2-3 months is the approximate
   goal.
 
-## Your organization is currently a committer
+## Your organization is currently a maintainer
 
-* First decide whether your organization really needs more people with commit access. Valid reasons
-  are "blast radius", a large organization that is working on multiple unrelated projects, etc.
-* Contact a senior committer for your organization and express interest.
-* Start doing PRs and code reviews under the guidance of your senior committer.
-* After a period of 1-2 months the existing senior committers will discuss granting "standard"
-  commit access.
-* "Standard" commit access can be upgraded to "senior" commit access after another 1-2 months of
-  work and another conference of the existing senior committers.
+* First decide whether your organization really needs more people with maintainer access. Valid
+  reasons are "blast radius", a large organization that is working on multiple unrelated projects,
+  etc.
+* Contact a senior maintainer for your organization and express interest.
+* Start doing PRs and code reviews under the guidance of your senior maintainer.
+* After a period of 1-2 months the existing senior maintainers will discuss granting "standard"
+  maintainer access.
+* "Standard" maintainer access can be upgraded to "senior" maintainer access after another 1-2
+  months of work and another conference of the existing senior maintainers.
+
```

Comment:
As we scale, I think we should aim to have a more even load balancing across maintainers of routine reviews, mailing list and Slack activity. We've all agreed this is desirable. I'm not sure if I have anything actionable for this review, so just consider this comment a friendly ping while we're thinking about governance and organizational structure.

The other thing to consider is what scenarios lead to someone no longer being a maintainer. I have no opinions on this other than to ask what are the best practices? Internally, folks gets the moral equivalent of maintainer when they are actively contributing and reviewing, but if they are no longer active and interested, it's removed.

Reply:
>As we scale, I think we should aim to have a more even load balancing across maintainers of routine reviews, mailing list and Slack activity. We've all agreed this is desirable. I'm not sure if I have anything actionable for this review, so just consider this comment a friendly ping while we're thinking about governance and organizational structure.

Yes I agree. I'm not really sure how to codify this. It's hard to hold people to doing this unless we actually start an "on-call" which is something that we could consider.

> The other thing to consider is what scenarios lead to someone no longer being a maintainer. I have no opinions on this other than to ask what are the best practices? Internally, folks gets the moral equivalent of maintainer when they are actively contributing and reviewing, but if they are no longer active and interested, it's removed.

In spirit, this is what most projects do, though again it will be hard to have a rule. For example I think FreeBSD has some rule where if a maintainer does not commit anything for 1 year they get removed. That's a very different type of project and that seems pretty long. I will try to come up with some basic text here.